frontends/web/src/components/forms/label.tsx

```diff
@@ -14,17 +15,18 @@
  * limitations under the License.
  */
 
-import { h } from 'preact';
+import { h, JSX } from 'preact';
+import * as style from './label.css';
 
-export default function Label({
+export function Label({
     className,
-    style,
     children,
+    id,
```

Comment:
why not drop this and use `for=...`? would be more intuitive to use the label props for everything.

Reply:
Agreed, but unfortunately `for` is a reserved word in JavaScript and Typescript goes nuts, therefore JSX uses `htmlFor`, see react https://reactjs.org/docs/dom-elements.html#htmlfor

In preact 8 there are only typings for `for`, but `htmlFor` typings are missing: `Property 'htmlFor' does not exist on type 'HTMLAttributes'.ts(2322)` and got introduced in preact 10.x, see https://github.com/preactjs/preact/commit/5046b75a6aa76c2cc1312ab81679b07827f15812#diff-814abe9ce1d0e89a7fa1ef941142487296d1423ffad605dc1188b544056c2bcfR540

So either keep it with a TODO comment or temporary reassign `for` to `htmlFor` with a comment to change it to `htmlFor`?

```diff
diff --git a/frontends/web/src/components/forms/label.tsx b/frontends/web/src/components/forms/label.tsx
index 2190577a..345e786c 100644
--- a/frontends/web/src/components/forms/label.tsx
+++ b/frontends/web/src/components/forms/label.tsx
@@ -21,12 +21,12 @@ import * as style from './label.css';
 export function Label({
     className,
     children,
-    id,
+    for: htmlFor, // TODO: change to htmlFor when mirgated away from preact@8.x
     ...props
 }: JSX.IntrinsicElements['label']) {
     const classes = [style.label, className].join(' ');
     return (
-        <label for={id} className={classes} {...props}>
+        <label for={htmlFor} className={classes} {...props}>
             {children}
         </label>
     );
```

Reply:
k didn't realize it was so complicated ;) i leave it to you what is best, anything is good for me.